pkg/commands/git_commands/working_tree.go

```diff
@@ -407,3 +408,26 @@ func (self *WorkingTreeCommands) ResetMixed(ref string) error {
 
 	return self.cmd.New(cmdArgs).Run()
 }
+
+func (self *WorkingTreeCommands) ObjectIDAtStage(path string, stage int) (string, error) {
+	cmdArgs := NewGitCmd("rev-parse").
+		Arg(fmt.Sprintf(":%d:%s", stage, path)).
+		ToArgv()
+
+	output, err := self.cmd.New(cmdArgs).RunWithOutput()
+	if err != nil {
+		return "", err
+	}
+
+	return strings.TrimSpace(output), nil
+}
+
+func (self *WorkingTreeCommands) MergeFile(strategy string, oursID string, baseID string, theirsID string) (string, error) {
+	cmdArgs := NewGitCmd("merge-file").
+		Arg("--object-id").
+		Arg(strategy).
+		Arg("-p", oursID, baseID, theirsID).
```

Comment:
Nitpicks: would use `--stdout` instead of `-p` for better readability, and put it on a line by itself; otherwise it looks like it is an option that takes the object ids as arguments.

```suggestion
		Arg("--stdout").
		Arg(oursID, baseID, theirsID).
```